#### Задание 1:
Даны 4 последовательности.

![task1](img/hw2.3.task1.PNG)

In [1]:
def a(n):
    return 2**n - n

In [2]:
def b(n):
    return 1/(1-n)

In [3]:
def c(n):
    return -1*1**n + (2*n)**(1/2)

In [4]:
def d(n):
    return (-1)**(2*n) + 1/n**2

In [5]:
def getList(l, n):
    return list(map(lambda x: l(x), n))

In [6]:
an = getList(a, list(range(1, 10)))
bn = getList(b, list(range(2, 10)))
cn = getList(c, list(range(1, 10)))
dn = getList(d, list(range(1, 10)))

an, bn, cn, dn

([1, 2, 5, 12, 27, 58, 121, 248, 503],
 [-1.0,
  -0.5,
  -0.3333333333333333,
  -0.25,
  -0.2,
  -0.16666666666666666,
  -0.14285714285714285,
  -0.125],
 [0.41421356237309515,
  1.0,
  1.4494897427831779,
  1.8284271247461903,
  2.1622776601683795,
  2.4641016151377544,
  2.7416573867739413,
  3.0,
  3.2426406871192848],
 [2.0,
  1.25,
  1.1111111111111112,
  1.0625,
  1.04,
  1.0277777777777777,
  1.0204081632653061,
  1.015625,
  1.0123456790123457])

#### 1.a исследовать их на монотонность

Последовательность  называется монотонно возрастающей, если для любого n, x(n) < x(n+1)

Последовательность  называется монотонно убывающей, если для любого n, x(n) > x(n+1)

{An}: возрастающая

{Bn}: возрастающая

{Cn}: возрастающая

{Dn}: убывающая

#### 1.б исследовать на ограниченность

Последовательность называется ограниченной сверху, если существует число M такое, что каждый элемент рассматриваемой последовательности удовлетворяет неравенству Xn <= M.

Последовательность называется ограниченной снизу, если существует число m такое, что каждый элемент заданной последовательности удовлетворяет неравенству Xn >= m

{An}: ограничена снизу m = 1

{Bn}: ограничена снизу m = -1

{Cn}: ограничена снизу m = 0.41421

{Dn}: ограничена сверху M = 2

#### 1.c найти пятый по счету член.

In [7]:
print('A(5):',a(5))
print('B(5):',b(6))
print('C(5):',c(5))
print('D(5):',d(5))

A(5): 27
B(5): -0.2
C(5): 2.1622776601683795
D(5): 1.04


#### Задание 2:
Найти 12-й член заданной неявно последовательности

![task2](img/hw2.3.task2.PNG)

In [8]:
def a(n):
    if n == 1:
        return 128
    else:
        return a(n-1) + 6

In [14]:
a(12)

194